ocrspace/main.py

```diff
@@ -57,21 +78,44 @@ def _parse(self, raw):
             raise Exception(raw['ErrorMessage'][0])
         return raw['ParsedResults'][0]['ParsedText']
 
-
-    def ocr_file(self, fp):
+    def query_api(self, url_data=None, image_file=None):
         """
-        Process image from a local path.
-        :param fp: A path or pointer to your file
+        Process the provided parameter.
+        :param url_data: Either an Image url or base64image
+        :param image_file: A path or file pointer to the image file
         :return: Result in JSON format
+        :raise: request.exceptions or general Exception
         """
-        with (open(fp, 'rb') if type(fp) == str else fp) as f:
+
+        if image_file:
             r = requests.post(
                 self.endpoint,
-                files={'filename': f},
+                headers={'apikey': self.api_key},
+                files={'filename': image_file},
                 data=self.payload,
+                timeout=30
             )
+        elif url_data:
+            r = requests.post(
+                self.endpoint,
+                headers={'apikey': self.api_key},
+                data=url_data,
+                timeout=30
+            )
+        else:
+            raise TypeError("either image_file or url_data must be provided")
+        r.raise_for_status()
         return self._parse(r.json())
 
+    def ocr_file(self, fp):
+        """
+        Process image from a local path.
+        :param fp: A path or pointer to your file
+        :return: Result in JSON format
+        """
+        with (open(fp, 'rb') if type(fp) == str else fp) as f:
```

Comment:
This is kind of dense, can we reorganize to be a bit clearer?

Reply:
can you clarify on this. Do you want to reorganize the code or the documentation? and if so how do you want to reorganize the code or documentation? Dense doesn't really communicate what you want to do

Reply:
> can you clarify on this. Do you want to reorganize the code or the documentation? and if so how do you want to reorganize the code or documentation? Dense doesn't really communicate what you want to do

The code could be made more readable (or more verbose).